XenAdmin/Alerts/Types/ClientUpdateAlert.cs

```diff
@@ -146,5 +147,50 @@ public static void DownloadAndInstallNewClient(ClientUpdateAlert updateAlert, IW
                     dlg.ShowDialog(parent);
             }
         }
+
+        public static void DownloadSource(IWin32Window parent)
+        {
+            var saveSourceDialog = new SaveFileDialog();

```

Comment:
It's a goog practice to construct and launch modal dialogs within a using directive so it gets disposed after use and releases OS resources.